syntax = "proto3";

message Empty {
}

message State {
  string state = 1;
}

message Goal {
  string goal = 1;
}

message Operators {
  repeated string ops = 1;
}

message Action {
  string action = 1;
}

message Result {
  string result = 1;
}

message Solution {
  // 0 = solvable
  // 1 = unsolvable
  int32 status = 1;
  repeated string action = 2;
}

message Problem {
  string problem = 1;
}

service AiddlExecutor {
  rpc assembleProblem(Goal) returns (Problem);
  rpc doNextAction(Empty) returns (Action);
  rpc processPlanningResult(Solution) returns (Result);
  rpc processActionResult(Result) returns (Empty);
  rpc processState(State) returns (Empty);
  rpc setOperators(Operators) returns (Empty);
}
